### ISSUE TITLE
Savon::SOAPFault Error when using wasabi 3.6.1

### ISSUE THREAD

**Issue by dantelljohann** [savonrb/savon#935]
# Bug report

**Current behavior:**
We are currently using `savon` version `2.12.1` and tried to update `wasabi` to `3.6.1`. When we did that our request that do:
```ruby
      Savon.client do
        ...
      end.call
```
Throw the following error:
```
Savon::SOAPFault
(soap:Client) Message part handleInteractiveOrder was not recognized.  (Does it exist in service WSDL?)
```

**Steps to reproduce current behavior:**
Ruby code that follows the pattern:
```ruby
Savon.client do
        ...
      end.call
```

**Expected behavior:**
Before updating wasabi to `3.6.1` we were on `3.5.0` and this error wasn't occurring. We would expect the same to happen with `3.6.1` since the dependency is listed as `wasabi ~> 3.4`

**System information:**

* ruby version: 2.6.5p114
* savon version: 2.12.0

**Comment by chastell** [comment]
Here’s a minimal reproducing example that uses a public WSDL:

```ruby
require 'savon'
wsdl = 'https://graphical.weather.gov/xml/DWMLgen/wsdl/ndfdXML.wsdl'
locals = {
  message: { listLatLon: '44.4437648,-121.1897322', product: 'time-series' },
}
Savon.client(wsdl: wsdl).call(:ndf_dgen_lat_lon_list, locals)
```

This runs without excepions on Savon 2.12.1 and Wasabi 3.5.0 but blows up on Savon 2.12.1 and Wasabi 3.6.1.

Also cross-linking #936 here, as I believe it’s the same issue.

**Comment by chastell** [comment]
My debugging led me to [this commit](https://github.com/savonrb/wasabi/commit/6fc97c6048b28aa79d053dc7bf9b736513c42c11) in Wasabi, which added [these lines](https://github.com/savonrb/wasabi/blame/a62be315efae1334c7dd85dcf62b823850321b15/lib/wasabi/parser.rb#L285-L296) to `parser.rb` and changed the Wasabi’s format, hope that helps. 😅

**Comment by zjeraar** [comment]
Yep, I have the same problem. All of the sudden got errors with some automated calls and could lead it back to the Wasabi upgrade from 3.5.0 to 3.6.1. Pinned wasabi to 3.5.0 for now and everything works again.

**Comment by nijave** [comment]
It seems like wasabi 3.6.1 is unnecessarily nesting things. To go along with @dantelljohann original post

Wasabi 3.5.0
![image](https://user-images.githubusercontent.com/384474/101692538-5ac25280-3a3e-11eb-9405-de95034697f5.png)

Wasabi 3.6.1
![image](https://user-images.githubusercontent.com/384474/101692557-601f9d00-3a3e-11eb-8b98-ec5742825bff.png)

So then this causes the tag namespacing code in savon to incorrectly omit the namespace (and then the receiving service throws a client error cause it doesn't know what you're talking about)

**Comment by olleolleolle** [comment]
@nijave Thanks for digging more! Any PRs to try and amend the nesting issues are welcome.

**Comment by diesl** [comment]
Not stale

**Comment by pcai** [comment]
This may have been a regression caused by https://github.com/savonrb/wasabi/pull/103 which was released in wasabi 3.7.0. Can you please reopen with additional information if this is still not working with wasabi 3.7+?

**Comment by diesl** [comment]
I can confirm it works again with a current version of wasabi 👍
